xtable-core/src/main/java/org/apache/xtable/parquet/ParquetPartitionHelper.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.xtable.parquet;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.xtable.model.schema.InternalPartitionField;
+import org.apache.xtable.model.schema.InternalSchema;
+import org.apache.xtable.model.schema.PartitionTransformType;
+import org.apache.xtable.model.stat.PartitionValue;
+import org.apache.xtable.model.stat.Range;
+import org.apache.xtable.schema.SchemaFieldFinder;
+
+public class ParquetPartitionHelper {
+  private static final ParquetPartitionHelper INSTANCE = new ParquetPartitionHelper();
+
+  public static ParquetPartitionHelper getInstance() {
+    return INSTANCE;
+  }
+
+  public List<InternalPartitionField> getInternalPartitionField(
+      Set<String> partitionList, InternalSchema schema) {
+    List<InternalPartitionField> partitionFields = new ArrayList<>();
+
+    for (String partitionKey : partitionList) {
+
+      partitionFields.add(
+          InternalPartitionField.builder()
+              .sourceField(SchemaFieldFinder.getInstance().findFieldByPath(schema, partitionKey))
+              .transformType(PartitionTransformType.VALUE)
+              .build());
+    }
+
+    return partitionFields;
+  }
+
+  // TODO logic is too complicated can be simplified
+  public List<PartitionValue> getPartitionValue(
+      String basePath,
+      String filePath,
+      InternalSchema schema,
+      Map<String, List<String>> partitionInfo) {
+    List<PartitionValue> partitionValues = new ArrayList<>();
+    java.nio.file.Path base = Paths.get(basePath).normalize();
```

Comment:
If you're always going to convert the basePath, you should try to find a way to convert it once in the caller and pass it in.

Reply:
Noted , overall i need to simplify the logic  as it is bit complicated